src/Particle/ParticleSet.h

```diff
@@ -19,682 +19,12 @@
-
-  ///SpeciesSet of particles
-  SpeciesSet my_species_;
-
-  ///Structure factor
-  std::unique_ptr<StructFact> structure_factor_;
-
-  ///multi walker structure factor data
-  ResourceHandle<SKMultiWalkerMem> mw_structure_factor_data_handle_;
-
-  /** map to handle distance tables
-   *
-   * myDistTableMap[source-particle-tag]= locator in the distance table
-   * myDistTableMap[ObjectTag] === 0
-   */
-  std::map<std::string, int> myDistTableMap;
-
-  /// distance tables that need to be updated by moving this ParticleSet
-  std::vector<std::unique_ptr<DistanceTable>> DistTables;
-
-  /// Descriptions from distance table creation.  Same order as DistTables.
-  std::vector<std::string> distTableDescriptions;
-
-  TimerList_t myTimers;
-
-  SingleParticlePos myTwist;
-
-  std::string ParentName;
-
-  ///total number of particles
-  size_t TotalNum;
-
-  ///array to handle a group of distinct particles per species
-  std::shared_ptr<Vector<int, OMPallocator<int>>> group_offsets_;
-
-  ///internal representation of R. It can be an SoA copy of R
-  std::unique_ptr<DynamicCoordinates> coordinates_;
-
-  /** compute temporal DistTables and SK for a new particle position
-   *
-   * @param iat the particle that is moved on a sphere
-   * @param newpos a new particle position
-   * @param maybe_accept if false, the caller guarantees that the proposed move will not be accepted.
-   */
-  void computeNewPosDistTables(Index_t iat, const SingleParticlePos& newpos, bool maybe_accept = true);
-
-
-  /** compute temporal DistTables and SK for a new particle position for each walker in a batch
-   *
-   * @param p_list the list of wrapped ParticleSet references in a walker batch
-   * @param iat the particle that is moved on a sphere
-   * @param new_positions new particle positions
-   * @param maybe_accept if false, the caller guarantees that the proposed move will not be accepted.
-   */
-  static void mw_computeNewPosDistTables(const RefVectorWithLeader<ParticleSet>& p_list,
-                                         Index_t iat,
-                                         const std::vector<SingleParticlePos>& new_positions,
-                                         bool maybe_accept = true);
-
-  /** actual implemenation for accepting a proposed move in forward mode
-   *
-   * @param iat the index of the particle whose position and other attributes to be updated
-   */
-  void acceptMoveForwardMode(Index_t iat);
-
-  /** reject a proposed move in forward mode
-   * @param iat the electron whose proposed move gets rejected.
-   */
-  void rejectMoveForwardMode(Index_t iat);
-
-  /// resize internal storage
-  inline void resize(size_t numPtcl)
-  {
-    TotalNum = numPtcl;
-
-    R.resize(numPtcl);
-    spins.resize(numPtcl);
-    GroupID.resize(numPtcl);
-    G.resize(numPtcl);
-    L.resize(numPtcl);
-    Mass.resize(numPtcl);
-    Z.resize(numPtcl);
-
-    coordinates_->resize(numPtcl);
-  }
-};
+using ParticleSet = ParticleSetT<QMCTraits::ValueType>;
```

Comment:
ParticleSet should be real-only.

Reply:
Is [MCWalkerConfiguration](https://github.com/QMCPACK/qmcpack/blob/864c22d19c26a8b41b0461bca5beb818cbbbd766/src/Particle/MCWalkerConfiguration.h#L55), which inherits from ParticleSet, real-only?

Reply:
it should.

Reply:
Since `ParticleSet` contains `GradType` and `QMCTraits::GradType` depends on `ValueType`. I don't see a straightforward way to make ParticleSet real-only.

https://github.com/QMCPACK/qmcpack/blob/864c22d19c26a8b41b0461bca5beb818cbbbd766/src/type_traits/QMCTypes.h#L44

Reply:
Given that in the current state in `develop` `ParticleSet` depends on `ValueType`, I think it falls outside the scope of this PR.